src/transformers/generation/utils.py

```diff
@@ -2597,6 +2597,15 @@ def _contrastive_search(
         unfinished_sequences = torch.ones(batch_size, dtype=torch.long, device=input_ids.device)
         model_kwargs = self._get_initial_cache_position(input_ids, model_kwargs)
 
+        # Create cosine_matrix_mask based on the attention_mask
+        cosine_matrix_mask = torch.ones_like(input_ids, dtype=torch.long)
+        if self.config.is_encoder_decoder:
+            if "decoder_attention_mask" in model_kwargs and model_kwargs["decoder_attention_mask"] is not None:
+                cosine_matrix_mask = model_kwargs["decoder_attention_mask"]
+        else:
+            cosine_matrix_mask = model_kwargs["attention_mask"]
+        cosine_matrix_mask = cosine_matrix_mask.repeat_interleave(top_k, dim=0)
+
```

Comment:
This code initializes a default mask and then updates it based on the model type.
- For encoder-decoder models, if `model_kwargs` contains a `decoder_attention_mask` (and it is not None), `cosine_matrix_mask` is set to this mask. If `decoder_attention_mask` is missing, it falls back to the default mask.
- For decoder-only models, `cosine_matrix_mask` is set to the `attention_mask` from `model_kwargs`.

Please let me know if there are any additional logic checks that need to be added